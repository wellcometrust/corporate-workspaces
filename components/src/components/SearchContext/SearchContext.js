// @flow
import React, { createContext, useState } from 'react';
import type { Node } from 'react';

const defaultState = {
  isSearchActive: false,
  openSearch: () => {}
};

type ContextProps = {|
  isSearchActive: boolean,
  openSearch: (bool: boolean) => void
|};

type ProviderProps = {|
  children: Node
|};

const SearchContext = createContext<ContextProps>(defaultState);

export const SearchContextProvider = ({ children }: ProviderProps) => {
  const [state, setState] = useState({
    isSearchActive: false,
    openSearch: bool =>
      setState(prevState => ({ ...prevState, isSearchActive: bool }))
  });

  return (
    <SearchContext.Provider value={state}>{children}</SearchContext.Provider>
  );
};

export default SearchContext;
