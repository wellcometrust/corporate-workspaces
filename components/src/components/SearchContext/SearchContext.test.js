import React from 'react';
import { mount } from 'enzyme';

import '../../test/mocks/match-media.mock';
import SearchContext, { SearchContextProvider } from './SearchContext';

describe('<SearchContextProvider />', () => {
  let output;

  beforeEach(() => {
    output = mount(
      <SearchContextProvider>
        <SearchContext.Consumer>
          {({ isSearchActive }) => (
            <>
              <div className="test-isSearchActive">
                {isSearchActive.toString()}
              </div>
            </>
          )}
        </SearchContext.Consumer>
      </SearchContextProvider>
    );
  });

  afterEach(() => {
    output.unmount();
  });

  it('renders the component', () => {
    expect(output).toMatchSnapshot();
  });

  it('has `isSearchActive` property set to false', () => {
    expect(output.find('.test-isSearchActive').text()).toBe('false');
  });
});
