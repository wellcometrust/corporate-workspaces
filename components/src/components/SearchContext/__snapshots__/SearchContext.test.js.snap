// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchContextProvider /> renders the component 1`] = `ReactWrapper {}`;
